trilium_alchemy/core/note/note.py

```diff
@@ -962,14 +952,46 @@ def __getitem__(self, key: str | int) -> str | Note:
             else:
                 return attr.value
 
-    def __setitem__(self, key: str | int, value_spec: ValueSpec):
+    def __setitem__(self, key: str, value_spec: ValueSpec):
         """
-        Create or update attribute with provided name.
+        Create or update attribute with provided name or index.
+
+        :param key: Attribute name
+        :param value_spec: Attribute value
         """
         self.attributes[key] = value_spec
 
-    def __contains__(self, key: str) -> bool:
-        return key in self.attributes._name_map
+    def __hash__(self):
+        return hash(self.note_id)
+
+    def __eq__(self, other):
+        if isinstance(other, Note):
+            return self.note_id == other.note_id
+        else:
+            return False
+
+    def __delitem__(self, key: str):
+        """
+        Delete owned attributes with provided name.
+
+        :param key: Attribute name
+        :raises KeyError: No such attribute
+        """
+        del self.attributes.owned[key]

```

Comment:
Nice, this is cleaner than repeating the looping logic.

Reply:
Agreed. With tests I was able to determine this worked as well.